python/ray/external_storage.py

```diff
@@ -272,20 +303,21 @@ def restore_spilled_objects(self, object_refs: List[ObjectRef],
 
     def delete_spilled_objects(self, urls: List[str]):
         for url in urls:
-            filename = parse_url_with_offset(url.decode()).base_url
-            os.remove(os.path.join(self.directory_path, filename))
+            path = parse_url_with_offset(url.decode()).base_url
+            os.remove(path)
```

Comment:
How come we don't need the os.path.join now?

Reply:
```
>>> a = "/abc/def"
>>> b = "/abc/def/a.py"
>>> os.path.join(a, b)
'/abc/def/a.py'
```
It looks like `os.path.join` automatically dedupes them. 

Our url contains the full path anyway, so we don't need os.path.join.